### DISCUSSION TITLE
High disk IO, resulting in k3s reboot

### DISCUSSION THREAD

**Discussion by Heyz6** [k3s-io/k3s#6605] [Q&A]
v1.21.13+k3s1

k3s instal agrs:
INSTALL_K3S_SKIP_DOWNLOAD=true INSTALL_K3S_EXEC='server --docker --cluster-init  --disable servicelb,traefik,local-storage,metrics-server --kubelet-arg=eviction-hard=memory.available<500Mi,nodefs.available<1Gi,imagefs.available<100Mi,nodefs.inodesFree<1Mi --kubelet-arg=image-gc-high-threshold=100' K3S_TOKEN=SECRET K3S_NODE_NAME="sqlinstall" ./k3s-install.sh

Using embedded etcd

Manufacturing high IO：
dd if=/dev/zero of=test bs=100M count=100

Please see the attached log for details
Dec 02 14:52:20 sqlinstall systemd[1]: k3s.service failed.
Dec 02 14:52:25 sqlinstall systemd[1]: k3s.service holdoff time over, scheduling restart.
Dec 02 14:52:25 sqlinstall systemd[1]: Stopped Lightweight Kubernetes.
[journalctl-k3s-log.txt](https://github.com/k3s-io/k3s/files/10138827/journalctl-k3s-log.txt)

I don't want high disk IO to cause k3s to reboot, how do I optimize

**Comment by brandond** [comment]
What kind of disk is this on? If you're running on SD cards or a very slow rotational disk, etcd (--cluster-init) is probably not for you; you should stick with sqlite, or put the etcd datastore on a second physical disk with higher iops.

Also, 1.21 is end of life; you should use 1.23 or newer.

**Comment by Heyz6** [comment]
It's a normal hard drive, not an SSD, but the same server, if I install kubernetes to test, there is no etcd or apiserver timeout, does k3s support inline etcd parameter tuning optimization about high IO scenarios, I don't want to restart the k3s process, which will cause some of my scripts to not work properly such as getting pod status through kubectl
I will test this on version 1.23 and hope to avoid this situation @brandond

**Comment by Heyz6** [comment]
oh,no Installed v1.25.4+k3s1, but the problem still can't be solved, whether there are parameters to optimize k3s installation @brandond

**Comment by brandond** [comment]
Using a newer version will not solve the problem of etcd not working well on slow rotational storage. I would suggest that you either move the K3s etcd datastore to SSD, or rebuild the cluster with a single-node sqlite datastore, which is less sensitive to disk latency.
